core/state-machine/src/overlayed_changes.rs

```diff
@@ -104,61 +306,68 @@ impl OverlayedChanges {
 		true
 	}
 
+	#[cfg(test)]
+	/// To allow value without extrinsic this can be use in test to disable change trie.
+	pub(crate) fn remove_changes_trie_config(&mut self) -> Option<ChangesTrieConfig> {
+		self.changes_trie_config.take()
+	}
+
+
 	/// Returns a double-Option: None if the key is unknown (i.e. and the query should be refered
 	/// to the backend); Some(None) if the key has been deleted. Some(Some(...)) for a key whose
 	/// value has been set.
 	pub fn storage(&self, key: &[u8]) -> Option<Option<&[u8]>> {
-		self.prospective.top.get(key)
-			.or_else(|| self.committed.top.get(key))
-			.map(|x| x.value.as_ref().map(AsRef::as_ref))
+		if let Some(overlay_value) = self.changes.top.get(key) {
+			if let Some(o_value) = overlay_value.get(self.changes.history.as_ref()) {
+				return Some(o_value.value.as_ref().map(|v| v.as_slice()))
+			}
+		}
+		None
 	}
 
 	/// Returns a double-Option: None if the key is unknown (i.e. and the query should be refered
 	/// to the backend); Some(None) if the key has been deleted. Some(Some(...)) for a key whose
 	/// value has been set.
 	pub fn child_storage(&self, storage_key: &[u8], key: &[u8]) -> Option<Option<&[u8]>> {
-		if let Some(map) = self.prospective.children.get(storage_key) {
-			if let Some(val) = map.get(key) {
-				return Some(val.value.as_ref().map(AsRef::as_ref));
-			}
-		}
-
-		if let Some(map) = self.committed.children.get(storage_key) {
-			if let Some(val) = map.get(key) {
-				return Some(val.value.as_ref().map(AsRef::as_ref));
+		if let Some(map) = self.changes.children.get(storage_key) {
+			if let Some(overlay_value) = map.get(key) {
+				if let Some(o_value) = overlay_value.get(self.changes.history.as_ref()) {
+					return Some(o_value.value.as_ref().map(|v| v.as_slice()))
+				}
 			}
 		}
-
 		None
 	}
 
 	/// Inserts the given key-value pair into the prospective change set.
 	///
 	/// `None` can be used to delete a value specified by the given key.
-	pub(crate) fn set_storage(&mut self, key: Vec<u8>, val: Option<Vec<u8>>) {
+	pub fn set_storage(&mut self, key: Vec<u8>, value: Option<Vec<u8>>) {
+		self.operation_from_last_gc += DEFAULT_GC_CONF.operation_cost(value.as_ref());
```

Comment:
What is the reason of having a dedicated structure if we access it as a global constant anyway? Should it be actually a field on `self`?

Reply:
I started to write the code parameterize-able so I kept the structure for if/when it needs to become parameterize-able.